src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,190 @@
+import net.sf.jabref.logic.importer.IdBasedParserFetcher;
+import net.sf.jabref.logic.importer.Parser;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.SearchBasedFetcher;
+import net.sf.jabref.logic.importer.fileformat.MedlineImporter;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Fetch or search from Pubmed http://www.ncbi.nlm.nih.gov/sites/entrez/
+ */
+public class MedlineFetcher implements IdBasedParserFetcher, SearchBasedFetcher {
+
+    private static final Log LOGGER = LogFactory.getLog(MedlineFetcher.class);
+
+    private static final Pattern ID_PATTERN = Pattern.compile("<Id>(\\d+)</Id>");
+    private static final Pattern COUNT_PATTERN = Pattern.compile("<Count>(\\d+)<\\/Count>");
+    private static final Pattern RET_MAX_PATTERN = Pattern.compile("<RetMax>(\\d+)<\\/RetMax>");
+    private static final Pattern RET_START_PATTERN = Pattern.compile("<RetStart>(\\d+)<\\/RetStart>");
+
+    private int count;
+    private int retmax;
+    private int retstart;
+
+    /**
+     * Removes all comma's in a given String
+     *
+     * @param query input to remove comma's
+     * @return input without comma's
+     */
+    private static String replaceCommaWithAND(String query) {
+        return query.replaceAll(", ", " AND ").replaceAll(",", " AND ");
+    }
+
+    /**
+     * When using 'esearch.fcgi?db=<database>&term=<query>' we will get a list of ID's matching the query.
+     * Input: Any text query (&term)
+     * Output: List of UIDs matching the query
+     *
+     * @see <a href="https://www.ncbi.nlm.nih.gov/books/NBK25500/">www.ncbi.nlm.nih.gov/books/NBK25500/</a>
+     */
+    private List<String> getPubMedIdsFromQuery(String query, int start, int pacing) throws FetcherException {
+        boolean doCount = true;
+        List<String> idList = new ArrayList<>();
+        try {
+            URL ncbi = createSearchUrl(query, start, pacing);
+            BufferedReader in = new BufferedReader(new InputStreamReader(ncbi.openStream()));
+            String inLine;
+            while ((inLine = in.readLine()) != null) {
+
+                Matcher idMatcher = ID_PATTERN.matcher(inLine);
+                if (idMatcher.find()) {
+                    idList.add(idMatcher.group(1));
+                }
+                Matcher retMaxMatcher = RET_MAX_PATTERN.matcher(inLine);
+                if (retMaxMatcher.find()) {
+                    retmax = Integer.parseInt(retMaxMatcher.group(1));
+                }
+                Matcher retStartMatcher = RET_START_PATTERN.matcher(inLine);
+                if (retStartMatcher.find()) {
+                    retstart = Integer.parseInt(retStartMatcher.group(1));
+                }
+                Matcher countMatcher = COUNT_PATTERN.matcher(inLine);
+                if (doCount && countMatcher.find()) {
```

Comment:
Why not moving "doCount" one level up? Just do countMatcher if `doCount`. Further, you should comment that you only count at the first hit when count results are there, because ...can-you-give-a-reason...?